phoenicis-multithreading/src/main/java/org/phoenicis/multithreading/ControlledThreadPoolExecutorService.java

```diff
@@ -48,6 +48,7 @@ public ControlledThreadPoolExecutorService(String name, int numberOfThread, int
     @Override
     public void execute(Runnable runnable) {
         try {
+            this.setCorePoolSize(numberOfThreads);
```

Comment:
I don't like to call a setter method with a constant during every `execute` invocation this just looks wrong to me.
What about https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ThreadPoolExecutor.html#allowCoreThreadTimeOut(boolean) can we use this alternatively?

Reply:
You are correct. Will change it in a new PR